action/container/ssh.go

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: MIT
+
+// Package container for dealing with containers via dagger
+package container
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"math/rand"
+	"strings"
+	"sync"
+
+	"dagger.io/dagger"
+)
+
+// ErrParseAddress is raised when containers IP address could not be parsed
+var ErrParseAddress = errors.New("could not parse address string")
+
+// OptsOpenSSH stores options for SSH tunnel for OpenSSH function
+type OptsOpenSSH struct {
+	WaitFunc  func()      // Waiting function holding container with SSH running
+	Password  string      // Filled in by OpenSSH function
+	IPv4      string      // Filled in by OpenSSH function
+	Port      string      // Filled in by OpenSSH function
+	MutexData *sync.Mutex // Mutex for modifying data in this struct
+
+	// We could do with single channel here, but for clarity and less mental overhead there are 2
+	TunnelClose chan (bool) // Channel to signal that SSH tunnel is ready
+	TunnelReady chan (bool) // Channel to signal that SSH tunnel is not longer needed and can be closed
+}
+
+// Wait calls WaitFunc
+func (s OptsOpenSSH) Wait() {
+	s.WaitFunc()
+}
+
+// Address function parses provided string and populates IPv4 and Port (port defaults to 22 if not found)
+func (s *OptsOpenSSH) Address(address string) error {
+	s.MutexData.Lock()
+	var err error
+	err = nil
```

Comment:
You can drop this line, Go always initializes with the zero/nil value of the underlying datatype. So the line above is sufficient here.